clients/src/main/java/org/apache/kafka/clients/producer/RecordMetadata.java

```diff
@@ -51,6 +52,11 @@ private RecordMetadata(TopicPartition topicPartition, long offset, long timestam
         this.topicPartition = topicPartition;
     }
 
+    @Deprecated
+    public RecordMetadata(TopicPartition topicPartition, long baseOffset, long relativeOffset) {
+        this(topicPartition, baseOffset, relativeOffset, Record.NO_TIMESTAMP, -1, -1, -1);
```

Comment:
I agree we should keep it, I was just unsure if semantics had been maintained. Seems like it's fine. Thanks for the references.